src/uct/tcp/tcp_ep.c

```diff
@@ -417,16 +406,43 @@ static UCS_CLASS_CLEANUP_FUNC(uct_tcp_ep_t)
 
     uct_tcp_ep_cleanup(self);
     uct_tcp_cm_change_conn_state(self, UCT_TCP_EP_CONN_STATE_CLOSED);
-    uct_tcp_ep_addr_cleanup(&self->peer_addr);
 
     ucs_debug("tcp_ep %p: destroyed on iface %p", self, iface);
 }
 
 UCS_CLASS_DEFINE(uct_tcp_ep_t, uct_base_ep_t);
 
-UCS_CLASS_DEFINE_NAMED_NEW_FUNC(uct_tcp_ep_init, uct_tcp_ep_t, uct_tcp_ep_t,
-                                uct_tcp_iface_t*, int,
-                                const struct sockaddr_storage*)
+ucs_status_t uct_tcp_ep_init(uct_tcp_iface_t *iface, int fd,
+                             const struct sockaddr *dest_addr,
+                             uct_tcp_ep_t **ep_p)
+{
+    ucs_status_t status;
+    uct_tcp_ep_t *ep;
+
+    ep = ucs_malloc(sizeof(uct_tcp_ep_t) + iface->config.sockaddr_len,
+                    "tcp_ep");
+    if (ep == NULL) {
+        ucs_error("failed to allocate tcp ep");
+        status = UCS_ERR_NO_MEMORY;
+        goto err;
+    }
+
+    status = UCS_CLASS_INIT(uct_tcp_ep_t, ep, iface, fd, dest_addr);
+    if (status != UCS_OK) {
+        ucs_error("failed to initialize tcp ep, status;%s",
+                  ucs_status_string(status));
```

Comment:
no need to print an error here

Reply:
unresolved

Reply:
@yosefe I removed this log message.  is anything more to do here?

Reply:
right - missed it